homeassistant/components/reolink/host.py

```diff
@@ -363,32 +364,52 @@ async def handle_webhook(
         self, hass: HomeAssistant, webhook_id: str, request: Request
     ):
         """Shield the incoming webhook callback from cancellation."""
-        await asyncio.shield(self.handle_webhook_shielded(hass, webhook_id, request))
+        self._webhook_read_done.clear()
+        shielded_task = hass.async_create_task(
+            self._handle_webhook(hass, webhook_id, request)
+        )
+        asyncio.shield(shielded_task)
```

Comment:
If you don't await the task there is no risk of cancelation propagating to the task because its now being run by the event loop